importers/audio_object.py

```diff
@@ -30,13 +36,92 @@ async def import_tracks(key: str):
     """
     print(f"Retrieving release info with key {key} from Muziekweb")
     # Get data from Muziekweb
-    tracks, music_works, persons = get_mw_audio_1track(key)
+    audio_objects, music_recordings, music_works, persons, music_groups = get_mw_audio_1track(key)
     # tracks = get_mw_audio(key)
 
-    if tracks is None or len(tracks) == 0:
+    if audio_objects is None or len(audio_objects) == 0:
         print(f"No track data received for {key}")
         return
 
+    #####################################
+    # MUSICCOMPOSITION
+    # Loop the music works to create the CE_MusicComposition on the CE
+    #####################################
+    for work in music_works:
+
+        work.identifier = await lookupIdentifier("MusicComposition", work.source)
+
+        if work.identifier is not None:
+            print(f"Updating work {work.identifier} in Trompa CE\n", end="")
+
+            response = await ce.connection.submit_query_async(mutation_update_music_composition(**work.as_dict()))
+            work.identifier = response["data"]["UpdateMusicComposition"]["identifier"]
+        else:
+            print("Inserting new work {} in Trompa CE\n".format(work.name))
+
+            response = await ce.connection.submit_query_async(mutation_create_music_composition(**work.as_dict()))
+            work.identifier = response["data"]["CreateMusicComposition"]["identifier"]
+
+    print(f"Importing music composition {work.identifier} done.\n")
+
+    #####################################
+    # MUSICRECORDING
+    # Loop the music recordings to create the CE_MusicRecording on the CE
+    #####################################
+    for recording in music_recordings:
```

Comment:
what's the situation when this is going to return more than 1 recording? From what I understand `get_mw_audio_1track` should only return one item, is that right?

Reply:
Indeed, but it was the logic used for the audioobjects, so I assumed that there was a logic behind doing this loop:
https://github.com/trompamusic/ce-data-import/blob/master/importers/audio_object.py#L108

Reply:
Juan confirmed me that to the main script for importing the tracks it can be passed an input file with a list of IDs to be imported
https://github.com/trompamusic/ce-data-import/blob/master/import-mw.py#L59

Then, I suppose this loop serves for that case

Reply:
but I think this method imports just a single item - the method is called "import_tracks", but the argument is called "Key", and the documentation says that it loads just one item. It also calls a method "get_mw_audio_1track" which sounds like it gets just one item. The question is why does get_mw_audio_1track return multiple items?
https://github.com/trompamusic/ce-data-import/blob/8ca397e414f4146b9b1c290b3055c69cd5c088b9/importers/audio_object.py#L179-L197
There may be a specific behaviour from the API that I don't know about. Perhaps there are multiple `<Track>` elements sometimes?